hypersync-net-types/tests/backward_compatibility.rs

```diff
@@ -0,0 +1,207 @@
+use hypersync_net_types::*;
+use pretty_assertions::assert_eq;
+use serde_json::Value;
+
+#[test]
+fn test_old_log_selection_deserialization() {
+    let old_format = r#"{
+        "address": ["0x6b175474e89094c44da98b954eedeac495271d0f"],
+        "topics": [
+            ["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"]
+        ]
+    }"#;
+
+    let log_selection: LogSelection =
+        serde_json::from_str(old_format).expect("Should be able to deserialize old format");
+
+    assert_eq!(log_selection.include.address.len(), 1);
+    assert_eq!(log_selection.include.topics.len(), 1);
+    assert!(log_selection.exclude.is_none());
+
+    // Test that serialized form can be deserialized back correctly
+    let serialized_val = serde_json::to_value(&log_selection).unwrap();
+    let original_val: Value = serde_json::from_str(old_format).unwrap();
+    assert_eq!(
+        serialized_val, original_val,
+        "Serializing new format without exclude should be the same as old format"
+    );
+}
```

Comment:
This just checks that old json queries deserialize properly to the new format.

And the new format withoud "exclude" filters, serializes the same way as the old format so they will still work with old hypersync versions.